File: setup-admin.py
#!/usr/bin/env python3
"""
Supabase Admin User Setup Script
Setup admin user for Reelanylizer

Usage:
    python setup-admin.py
    
Requirements:
    pip install supabase python-dotenv
"""

import os
import sys
import json
from pathlib import Path
from dotenv import load_dotenv

# Load environment variables
load_dotenv()
load_dotenv(".env.local")

try:
    from supabase import create_client
except ImportError:
    print("❌ Missing dependencies!")
    print("Install with: pip install supabase python-dotenv")
    sys.exit(1)

# Configuration
SUPABASE_URL = os.getenv("VITE_SUPABASE_URL")
SUPABASE_SERVICE_KEY = os.getenv("SUPABASE_SERVICE_KEY")

ADMIN_EMAIL = "owsmboy7383@gmail.com"
ADMIN_PASSWORD = "Aman@73550"

def validate_credentials():
    """Validate Supabase credentials"""
    if not SUPABASE_URL:
        print("❌ Missing VITE_SUPABASE_URL")
        print("   Set it in your .env file or export it:")
        print("   export VITE_SUPABASE_URL='your-url'")
        return False
    
    if not SUPABASE_SERVICE_KEY:
        print("❌ Missing SUPABASE_SERVICE_KEY")
        print("   Set it in your .env file or export it:")
        print("   export SUPABASE_SERVICE_KEY='your-service-key'")
        return False
    
    return True

def setup_admin():
    """Setup admin user in Supabase"""
    if not validate_credentials():
        sys.exit(1)
    
    try:
        print("🔐 Setting up admin user...\n")
        
        # Create Supabase client with service key
        supabase = create_client(SUPABASE_URL, SUPABASE_SERVICE_KEY)
        
        # Create admin user
        print(f"📧 Creating user: {ADMIN_EMAIL}")
        user_response = supabase.auth.admin.create_user({
            "email": ADMIN_EMAIL,
            "password": ADMIN_PASSWORD,
            "email_confirm": True
        })
        
        user_id = user_response.user.id
        print(f"✅ User created: {user_id}\n")
        
        # Assign admin role
        print(f"🔑 Assigning admin role...")
        role_response = supabase.table("user_roles").insert({
            "user_id": user_id,
            "role": "admin"
        }).execute()
        
        print(f"✅ Admin role assigned\n")
        
        # Print summary
        print("=" * 60)
        print("✨ ADMIN SETUP COMPLETE!")
        print("=" * 60)
        print(f"📧 Email:    {ADMIN_EMAIL}")
        print(f"🔒 Password: {ADMIN_PASSWORD}")
        print(f"🆔 User ID:  {user_id}")
        print("=" * 60)
        print(f"\n🚀 Login at: http://localhost:8080/bosspage-login")
        print(f"📊 Dashboard: http://localhost:8080/bosspage\n")
        
        return True
        
    except Exception as e:
        error_msg = str(e)
        
        if "already exists" in error_msg:
            print("⚠️  User already exists!")
            print("   You can login with the existing credentials")
            print(f"   Email: {ADMIN_EMAIL}")
            print(f"   Password: {ADMIN_PASSWORD}")
            return True
        else:
            print(f"❌ Error: {error_msg}")
            return False

if __name__ == "__main__":
    success = setup_admin()
    sys.exit(0 if success else 1)

File: setup-admin.js
/**
 * Admin Setup Script
 * This script creates an admin user in Supabase
 * Run with: node setup-admin.js
 */

import { createClient } from "@supabase/supabase-js";

const SUPABASE_URL = process.env.VITE_SUPABASE_URL;
const SUPABASE_SERVICE_KEY = process.env.SUPABASE_SERVICE_KEY || process.env.VITE_SUPABASE_SERVICE_KEY;

if (!SUPABASE_URL || !SUPABASE_SERVICE_KEY) {
  console.error("❌ Missing Supabase credentials!");
  console.error("Please set VITE_SUPABASE_URL and SUPABASE_SERVICE_KEY environment variables");
  process.exit(1);
}

const supabase = createClient(SUPABASE_URL, SUPABASE_SERVICE_KEY);

const adminEmail = "owsmboy7383@gmail.com";
const adminPassword = "Aman@73550";

async function setupAdmin() {
  try {
    console.log("🔐 Setting up admin user...");

    // Create admin user via Auth API
    const { data: authData, error: authError } = await supabase.auth.admin.createUser({
      email: adminEmail,
      password: adminPassword,
      email_confirm: true,
    });

    if (authError) {
      if (authError.message.includes("already exists")) {
        console.log("⚠️  Admin user already exists, attempting to update...");
        
        // Find existing user
        const { data: userData, error: userError } = await supabase.auth.admin.listUsers();
        if (userError) throw userError;
        
        const existingUser = userData.users.find((u) => u.email === adminEmail);
        if (!existingUser) {
          throw new Error("User not found and could not be created");
        }
        authData.user = existingUser;
      } else {
        throw authError;
      }
    }

    const userId = authData.user?.id;
    if (!userId) {
      throw new Error("Failed to get user ID");
    }

    console.log(`✅ Admin user created/found: ${userId}`);

    // Check if admin role already exists
    const { data: existingRole } = await supabase
      .from("user_roles")
      .select("*")
      .eq("user_id", userId)
      .eq("role", "admin");

    if (existingRole && existingRole.length > 0) {
      console.log("✅ Admin role already assigned");
    } else {
      // Insert admin role
      const { error: roleError } = await supabase.from("user_roles").insert({
        user_id: userId,
        role: "admin",
      });

      if (roleError) throw roleError;
      console.log("✅ Admin role assigned");
    }

    console.log("\n✨ Admin Setup Complete!");
    console.log("📧 Email: " + adminEmail);
    console.log("🔒 Password: " + adminPassword);
    console.log("\n🚀 Login at: http://localhost:8080/bosspage-login");
  } catch (error) {
    console.error("❌ Admin setup failed:", error.message);
    process.exit(1);
  }
}

setupAdmin();
